### PR TITLE
browser(firefox): support internal drag and drop

### PR DESCRIPTION
**Pull request by JoelEinbinder**
#1094

https://github.com/JoelEinbinder/juggler/commit/5686e43cb0315a056b6357ce95480000f7a719b4

### CODE REVIEW

File: browser_patches/firefox/patches/bootstrap.diff [hunk 1]
```diff
@@ -4415,8 +4428,61 @@ index 0000000000000000000000000000000000000000..63754937b3464794227fe894b3d6057f
...
++      0 /* clickCount */,
++      window.mozInnerScreenX + x,
++      window.mozInnerScreenY + y,
++      x,
++      y,
++      modifiers & 2 /* ctrlkey */,
++      modifiers & 1 /* altKey */,
++      modifiers & 4 /* shiftKey */,
++      modifiers & 8 /* metaKey */,
++      0 /* button */, // firefox always has the button as 0 on drops, regardless of which was pressed
```
Comment (yury-s) [inline browser_patches/firefox/patches/bootstrap.diff:4460]: Shouldn't it be actual button for `dragover` events?

File: browser_patches/firefox/patches/bootstrap.diff [hunk 2]
```diff
@@ -3711,6 +3711,12 @@ index 0000000000000000000000000000000000000000..63754937b3464794227fe894b3d6057f
 +
 +const {Helper} = ChromeUtils.import('chrome://juggler/content/Helper.js');
 +const {NetUtil} = ChromeUtils.import('resource://gre/modules/NetUtil.jsm');
++const dragService = Cc["@mozilla.org/widget/dragservice;1"].getService(
++  Ci.nsIDragService
++);
++const obs = Cc["@mozilla.org/observer-service;1"].getService(
++  Ci.nsIObserverService
```
Comment (yury-s) [inline browser_patches/firefox/patches/bootstrap.diff:3718]: nit: obs -> observerService ?

**Review by yury-s — APPROVED** [review]

File: browser_patches/firefox/patches/bootstrap.diff (hunk 1 above)
Comment (JoelEinbinder) [reply browser_patches/firefox/patches/bootstrap.diff:4460]: no, its always 0 for mousemove/dropover events as well.

File: browser_patches/firefox/patches/bootstrap.diff (hunk 2 above)
Comment (JoelEinbinder) [reply browser_patches/firefox/patches/bootstrap.diff:3718]: they use obs other places in their code. I firefox style makes it easy to see where this is copy pasted from.

**Pull request merged**